uwblueprint/project-read-frontend
# Refactor Field type to DefaultField and DynamicField

### Notion ticket link



N/A



### Implementation description

- refactoring the `Field` type into `DefaultField` and `DynamicField` in preparation for moving the reg form to TS
- moved FieldAPI.js → DynamicFieldAPI.ts



### Steps to test

1. run the backend with [#64](https://github.com/uwblueprint/project-read-backend/pull/64) if not yet merged to main
2. `yarn start`
3. ensure that the family table and reg form show up as expected (no functional changes)



### What should reviewers focus on?

- ts best practices & code style!

### Checklist
